itou/cities/tests.py

```diff
@@ -1,12 +1,205 @@
+                "codeRegion": "84",
+                "codesPostaux": ["01400"],
+                "nom": "L'Abergement-de-Varey",
+            },
+        ],
+    )
+
+    # Introduce a change in one of the arrondissements: the change is now permanent
+    # and any new automatic sync update to it will be skipped.
+    marseille_1er = City.objects.get(slug="marseille-1er-13")
+    marseille_1er.name = "Marssssssssseillle bébé"
+    marseille_1er.save()
+
+    management.call_command("sync_cities", wet_run=True)
+    stdout, stderr = capsys.readouterr()
+    assert stderr == ""
+    assert stdout.splitlines() == [
+        "count=3 label=City had the same key in collection and queryset",
+        "! skipping manually edited city=Marssssssssseillle bébé (13) from update",
+        "count=1 label=City added by collection",
+        '\tADDED {"centre": {"coordinates": [4.9306, 46.1517], "type": "Point"}, '
+        '"code": "01003", "codeDepartement": "01", "codeRegion": "84", '
+        '"codesPostaux": ["01400"], "nom": "L\'Abergement-de-Varey"}',
+        "count=1 label=City removed by collection",
+        "\tREMOVED L'Abergement-de-Varey (01)",
+        "> successfully deleted count=1 cities insee_codes={'01002'}",
+        "> successfully created count=1 new cities",
+        "> successfully updated count=0 cities",  # no update to post codes
+    ]
+
+    def to_dict(obj):
+        dct = obj.__dict__
+        del dct["id"]
+        del dct["_state"]
+        dct["coords"] = str(dct["coords"])
+        return dct
+
+    assert [to_dict(x) for x in City.objects.all().order_by("code_insee")] == [
+        to_dict(x)
+        for x in [
+            City(
+                code_insee="01001",
+                name="L'Abergement-Clémenciat",
+                slug="labergement-clemenciat-01",
+                department="01",
+                edition_mode="AUTO",
+                coords=Point(4.9306, 46.1517),
+                post_codes=["01234", "01400"],
+            ),
+            City(
+                code_insee="01003",
+                name="L'Abergement-de-Varey",
+                slug="labergement-de-varey-01",
+                department="01",
+                edition_mode="AUTO",
+                coords=Point(4.9306, 46.1517),
+                post_codes=["01400"],
+            ),
+            City(
+                code_insee="13201",
+                name="Marssssssssseillle bébé",
+                slug="marseille-1er-13",
+                department="13",
+                edition_mode="MANUAL",
+                coords=Point(5.3828, 43.3002),
+                post_codes=["13001"],
+            ),
+            City(
+                code_insee="13202",
+                name="Marseille 2e",
+                slug="marseille-2e-13",
+                department="13",
+                edition_mode="AUTO",
+                coords=Point(5.3496, 43.3225),
+                post_codes=["13002"],
+            ),
+        ]
```

Comment:
Autrement, pourquoi ne pas utiliser `assertQuerySetEqual` avec un `transform=` ?

Reply:
Ca me force à utiliser une classe `unittest` et donc de remplacer toutes les utilisations de mes fixtures pytest par autre chose, donc en gros à plus ou moins réécrire pour faire quelque chose d'essentiellement identique. J'ai essayé, j'ai fait machine arrière. A moins que je ne loupe un truc évident ?

Reply:
https://pytest-django.readthedocs.io/en/latest/helpers.html#assertions:

All of Django’s [TestCase](https://docs.djangoproject.com/en/stable/topics/testing/tools/#django.test.TestCase) [Assertions](https://docs.djangoproject.com/en/stable/topics/testing/tools/#assertions) are available in pytest_django.asserts, e.g.

```
from pytest_django.asserts import assertTemplateUsed
```

Reply:
TIL. Merci !